src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -97,11 +100,9 @@ private Layout waitForBootstrap() {
             try {
                 Thread.sleep(1000);
                 currentLayout = layoutServer.getCurrentLayout();
-            } catch (InterruptedException err) {
-                log.warn("Thread interrupted.");
-                if (isShutdown) {
-                    return null;
-                }
+            } catch (InterruptedException ie) {
```

Comment:
As per my understanding, the shutdownNow's best effort is sending an interrupt. If the thread fails to respond to this interrupt, it does not terminate.
Just to be safe, I can change the implementation to use a volatile flag.